app/app.go

```diff
@@ -34,11 +34,13 @@ type App struct {
 }
 
 // New returns a new App.
+// If ctx contains a zerolog logger it is used for logging.
+// adminPort must be a valid port number or it will fail silently.
 func New(ctx context.Context, adminPort string) (*App, error) {
```

Comment:
With the changes made, I believe `ctx` shoud be removed as a parameter and a `logger` should be passed instead. As it stands, it feels like context is being used to pass parameters, as it is only used to extract a logger and then discarded.

Passing a context will be misleading as it implies a wider range of possibilities. It's possible that this problem already existed, it's just more explicit now